runtime/src/launch/pbs-aprun/launch-pbs-aprun.c

```diff
@@ -352,16 +352,24 @@ static char** chpl_launch_create_argv(int argc, char* argv[],
     fprintf(stdout, "QSUB script written to '%s'\n", qsubOptions);
     return NULL;
   } else {
-  fprintf(expectFile, "\\n\"\n");
+    fprintf(expectFile, "\\n\"\n");
     fprintf(expectFile, "interact -o -ex \"$chpl_prompt\" {return}\n");
-    fprintf(expectFile, "send \"echo CHPL_EXIT_CODE:\\$?\\n\"\n");
+    fprintf(expectFile, "send \"set retval=$?\\n\"\n");
+    fprintf(expectFile, "send \"\\[ \\$retval = 0 \\] && echo \\\"JOB SUCCEEDED (done)\\\" || echo \\\"JOB FAILED: \\$retval (done)\\\"\\n\"\n");
+    // Being a bit excessive with the expects here
+    fprintf(expectFile, "expect -ex \"(done)\" {}\n");
+    fprintf(expectFile, "expect -ex \"(done)\" {}\n");
     fprintf(expectFile, "expect {\n");
-    fprintf(expectFile, "  -ex \"CHPL_EXIT_CODE:0\" {set exitval \"0\"}\n");
-    fprintf(expectFile, "  -re \"CHPL_EXIT_CODE:.\" {set exitval \"1\"}\n");
+    fprintf(expectFile, "    -ex \"JOB SUCCEEDED\" { set exitval \"0\" }\n");
+    fprintf(expectFile, "    -re \"JOB FAILED:.\" { set exitval \"-1\" }\n");
```

Comment:
I think it would be good to maintain the previous exit code of 1 when a job fails instead of changing it to -1 here.